python/dgl/heterograph_index.py

```diff
@@ -227,6 +227,33 @@ def copy_to(self, ctx):
         """
         return _CAPI_DGLHeteroCopyTo(self, ctx.device_type, ctx.device_id)
 
+    def shared_memory(self, name, ntypes=None, etypes=None, formats=('coo', 'csr', 'csc')):
+        """Return a copy of this graph in shared memory
+
+        Parameters
+        ----------
+        name : str
+            The name of the shared memory.
+        ntypes : list of str
+            Name of node types
+        etypes : list of str
+            Name of edge types
```

Comment:
Why do we need to provide `ntypes` and `etypes`?

Reply:
We want to store the whole heterogeneous graph object into shared memory so we can load it only using its name, which needs to store ntypes and etypes. Writing to shared memory is done in C++, through the HeteroGraphIndex interface. But a HeteroGraphIndex object does not contain ntypes and etypes. So we need to pass them as arguments.